sdk/appconfiguration/app-configuration/review/app-configuration.api.md

```diff
@@ -146,40 +215,79 @@ export type GetRevisionsResponse = KeyValueListResult & GetRevisionsHeaders & {
 };
 
 // @public
-export interface ListConfigurationSettingPage extends Pick<GetKeyValuesResponse, Exclude<keyof GetKeyValuesResponse, 'items'>> {
+export interface KeyValueListResult {
+    items?: ConfigurationSetting[];
+    nextLink?: string;
+}
+
+// @public
+export interface ListConfigurationSettingPage extends Pick<GetKeyValuesResponse, Exclude<keyof GetKeyValuesResponse, "items">> {
     items: ConfigurationSetting[];
 }
 
 // @public
-export interface ListConfigurationSettingsOptions extends Pick<AppConfigurationGetKeyValuesOptionalParams, Exclude<keyof AppConfigurationGetKeyValuesOptionalParams, 'key' | 'label' | 'select' | 'after'>> {
+export interface ListConfigurationSettingsOptions extends Pick<AppConfigurationGetKeyValuesOptionalParams, Exclude<keyof AppConfigurationGetKeyValuesOptionalParams, "key" | "label" | "select" | "after">> {
     fields?: (keyof ConfigurationSetting)[];
     keys?: string[];
     labels?: string[];
 }
 
 // @public
-export interface ListRevisionsOptions extends Pick<AppConfigurationGetRevisionsOptionalParams, Exclude<keyof AppConfigurationGetRevisionsOptionalParams, 'key' | 'label' | 'select' | 'after'>> {
+export interface ListRevisionsOptions extends Pick<AppConfigurationGetRevisionsOptionalParams, Exclude<keyof AppConfigurationGetRevisionsOptionalParams, "key" | "label" | "select" | "after">> {
     fields?: (keyof ConfigurationSetting)[];
     keys?: string[];
     labels?: string[];
 }
 
 // @public
-export interface ListRevisionsPage extends Pick<GetRevisionsResponse, Exclude<keyof GetRevisionsResponse, 'items'>> {
+export interface ListRevisionsPage extends Pick<GetRevisionsResponse, Exclude<keyof GetRevisionsResponse, "items">> {
     items: ConfigurationSetting[];
 }
 
-// Warning: (ae-forgotten-export) The symbol "PutKeyValueHeaders" needs to be exported by the entry point index.d.ts
-// 
 // @public
-export type PutKeyValueResponse = ConfigurationSetting & PutKeyValueHeaders & {
+export interface PutKeyValueHeaders {
+    eTag?: string;
+    syncToken?: string;
+}
+
+// @public
+type PutKeyValueResponse = ConfigurationSetting & PutKeyValueHeaders & {
     _response: coreHttp.HttpResponse & {
         parsedHeaders: PutKeyValueHeaders;
         bodyAsText: string;
         parsedBody: ConfigurationSetting;
     };
 };
 
+export { PutKeyValueResponse as AddConfigurationSettingResponse }
```

Comment:
hmm this pattern feels a bit weird. Would it make more sense to have real interfaces for these things that extend PetKeyValueResponse but don't add to it? e.g.

```ts
interface AddConfigurationSettingsResponse extends PutKeyValueResponse { }
```

Reply:
Tell me more about this weirdness? :)

I've been distinguishing between two cases:

1. I'm doing some sort of transformation or modification of the original type (for instance, I'm using Pick<> or variants). These get a interface/extends pattern.
2. I'm just renaming it because the original name is obviously very-generated so I'd rather give it a normal name, but it's otherwise unchanged.

It seems that for #2 it's nice to indicate that with less typing if possible.

Reply:
Yeah, I get that it's just an alias since there's nothing custom. It just seemed interesting to re-export that same interface multiple times with different names. I don't think I really have any strong feelings about it. 

I wish there was a way we could end up not having this 1-1 pattern for "every GET call gets its own interface for options", but I haven't come up with anything too clever yet.

Reply:
Do both methods show up in documentation the same way? That might be a potential reason to choose one or the other.

Reply:
@chradek submitted this issue already(#5013), which was part of the reason I started doing the interface/extends pattern. I need to figure out what I'm missing because even using the nice workaround he provided I still get dead links in my outputted docs. 

I'll come back in another PR and hit this again.